lib/dialects/abstract/query-generator.js

```diff
@@ -1979,25 +1980,31 @@ var QueryGenerator = {
 
           $baseKey = '('+$baseKey+')';
 
+          castKey = function ($item) {
```

Comment:
Yep, my bad. Committed an early version of this PR and not the refactored one .... :D